### PR TITLE
Add sanitization logic (isValidRuleTree(), isValidRuleConfig()) for rule tree

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Add sanitization logic (isValidRuleTree(), isValidRuleConfig()) for rule tree

Related issues: https://github.com/ainblockchain/ain-blockchain/issues/413

### CODE REVIEW

**Review by liayoo — APPROVED** [review]
LGTM with some minor comments. Thanks for the nice refactoring!

File: db/state-util.js [hunk 1]
```diff
@@ -331,24 +304,25 @@ function isValidOwnerConfig(ownerConfig) {
   return { isValid: true, invalidPath: '' };
 }
 
-function isValidOwnerTreeRecursive(ownerTree, path) {
-  if (!ChainUtil.isDict(ownerTree) || ChainUtil.isEmpty(ownerTree)) {
+function isValidConfigTreeRecursive(stateTree, path, configLabel, isValidStateConfig) {
```
[the remark this answers is not in the document]
Comment (platfowner) [reply db/state-util.js]: Done. Thanks for the nice idea!

File: unittest/state-util.test.js [hunk 2]
```diff
@@ -527,6 +529,131 @@ describe("state-util", () => {
...
+      assert.deepEqual(isValidRuleTree(null), {isValid: true, invalidPath: ''});
+      assert.deepEqual(isValidRuleTree({
+        '.write': true 
+      }), {isValid: true, invalidPath: ''});
+      assert.deepEqual(isValidRuleTree({
+        some_path1: {
+          '.write': true
+        },
+        some_path2: {
+          '.write': "auth.addr ==== 'abcd"
```
[the remark this answers is not in the document]
Comment (platfowner) [reply unittest/state-util.test.js]: Fixed. Thanks!

**Comment by platfowner** [comment]
Thanks for the fast review! @liayoo 
Being merged shortly..

**Pull request merged**